giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/zookeeper/impl/NepomukServiceDiscoverer.java

```diff
@@ -27,114 +27,121 @@
+		checkServices();
+	}
+
+	private void checkServices() {
+		List<String> uris;
+		try {
+			uris = curatorFramework.getChildren().forPath(znode);
+		} catch (Exception e) {
+			// this is actually throwing Exception
+			msgHandler.handleMessage("Could not get Nepomuk entries in Zookeeper.", e, MessageType.WARNING);
+			return;
+		}
+		for (String uri : uris) {
+			logger.debug("Found: " + uri);
+			byte[] url;
+			try {
+				url = curatorFramework.getData().forPath(ZKPaths.makePath(znode, uri));
+			} catch (Exception e) {
+				// method is actually throwing exception
+				msgHandler.handleMessage("Could not get Nepomuk URL from Zookeeper.", e, MessageType.ERROR);
+				return;
+			}
+			if (url != null) {
+				logger.debug("with URL: " + new String(url));
+				msgHandler.handleMessage("Nepomuk instance found.",
+						"Giles found a Nepomuk instance at ZNode " + uri + " with URL " + new String(url),
+						MessageType.INFO);
+			} else {
+				msgHandler.handleMessage("No Nepomuk URL configured.",
+						"There was no Nepomuk instance found at " + new String(uri), MessageType.WARNING);
+				logger.debug("There was no URL registered with the ZNode.");
+			}
+		}
+	}
+
+	@Override
+	public String getRandomNepomukInstance() throws NoNepomukFoundException {
+		// throws Exception
+		List<String> uris;
+		int randomInstance = 0;
+		try {
+			uris = curatorFramework.getChildren().forPath(znode);
+			if((uris != null) && (uris.size() > 0)) {
+			randomInstance = ThreadLocalRandom.current().nextInt(0, uris.size());
+			//error after randomInstance is generated
+			}
+		} catch (Exception e) {
+			throw new NoNepomukFoundException(e);
+		}
+		
+		
+		for (int i = 0; i < uris.size(); i++) {
+			// get random Nepomuk instance
+			byte[] urlBytes;
+			try {
+				urlBytes = curatorFramework.getData().forPath(ZKPaths.makePath(znode, uris.get(randomInstance)));
+			} catch (Exception e) {
+				throw new NoNepomukFoundException(e);
+			}
+			if (urlBytes != null) {
+				String url = new String(urlBytes);
+				try {
+					if (validateUrl(url)) {
+						// if Nepomuk instance url specified and nepomuk responds with ok
+						// return url
+						return url;
+					}
+				} catch (IOException e) {
+					msgHandler.handleMessage("Nepomuk URL could not be validated.", e, MessageType.WARNING);
+				}
+			}
+
+			// if there is no URL or URL is not valid, let's take the next instance
+			randomInstance++;
+			// if we're at the end of the list, start from its beginning
+			if (randomInstance >= uris.size()) {
+				randomInstance -= uris.size();
+			}
+		}
+
+		return null;
+	}
+
+	private boolean validateUrl(String url) throws IOException {
+		URL myURL = new URL(url);
+		HttpURLConnection myConnection = (HttpURLConnection) myURL.openConnection();
+
+		try {
+			if (myConnection.getResponseCode() == HttpStatus.OK.value()) {
+				return true;
+			} else {
+				return false;
+			}
+		} finally {
+			myConnection.disconnect();
+		}
+	}
```

Comment:
please fix indentation before I review this, it's very hard to see what you've actually changed.